fhir-validation/src/main/java/com/ibm/fhir/validation/FHIRValidator.java

```diff
@@ -156,26 +165,71 @@ public static FHIRValidator validator() {
         return new FHIRValidator();
     }
 
-    private static class ValidatingNodeVisitor extends FHIRPathDefaultNodeVisitor {
-        private FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
-        private EvaluationContext evaluationContext;
-        private boolean includeResourceAssertedProfiles;
-        private List<String> profiles;
-        private List<Issue> issues = new ArrayList<>();
+    /**
+     * Validate a list of profile references to ensure they are supported (known by the FHIR registry) and applicable
+     * (the type constrained by the profile is compatible with the resource being validated).
+     *
+     * <p>Resource-asserted profile references that are unknown, generate an issue with severity WARNING. Profile references
+     * passed in as arguments to the {@link FHIRValidator#validate(EvaluationContext, boolean, String...)} method, generate an
+     * issue with severity ERROR.
+     *
+     * <p>If a specific profile is known but incompatible with the resource being validated, then an issue with severity
+     * ERROR is generated.
+     *
+     * @param resourceNode
+     *     the resource node being validated by a FHIRValidator instance
+     * @param profileReferences
+     *     the list of profile references to validate
+     * @param resourceAsserted
+     *     indicates whether the profile references came from the resource or were explicitly passed in as arguments
+     * @param issues
+     *     the list of issues to add to
+     */
+    private static void validateProfileReferences(
+            FHIRPathResourceNode resourceNode,
+            List<String> profileReferences,
+            boolean resourceAsserted,
+            List<Issue> issues) {
+        Class<?> resourceType = resourceNode.resource().getClass();
+        for (String url : profileReferences) {
+            StructureDefinition profile = ProfileSupport.getProfile(url);
+            if (profile == null) {
+                issues.add(Issue.builder()
+                    .severity(resourceAsserted ? IssueSeverity.WARNING : IssueSeverity.ERROR)
+                    .code(IssueType.NOT_SUPPORTED)
+                    .details(CodeableConcept.builder()
+                        .text(string("Profile: " + url + " is not supported"))
+                        .build())
+                    .expression(string(resourceNode.path()))
+                    .build());
+            } else if (!ProfileSupport.isApplicable(profile, resourceType)) {
+                issues.add(Issue.builder()
+                    .severity(IssueSeverity.ERROR)
+                    .code(IssueType.INVALID)
+                    .details(CodeableConcept.builder()
+                        .text(string("Profile: " + url + " is not applicable to resource type: " + resourceType.getSimpleName()))
```

Comment:
pretty minor, but i like to put the variable in `'` just so whitespace stands out
```suggestion
                        .text(string("Profile: '" + url + "' is not applicable to resource type: " + resourceType.getSimpleName()))
```

Reply:
I guess its just a style difference though